backend/src/ml_space_lambda/app_configuration/lambda_functions.py

```diff
@@ -18,39 +18,234 @@
+                IAMStatementProperty.RESOURCE: [
+                    get_account_arn("iam", f"role/{env_variables[EnvVariable.EMR_EC2_ROLE_NAME]}"),
+                    get_account_arn("iam", f"role/{env_variables[EnvVariable.EMR_SERVICE_ROLE_NAME]}"),
+                ],
+            },
+            {
+                IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                IAMStatementProperty.ACTION: ["ec2:AuthorizeSecurityGroupIngress"],
+                IAMStatementProperty.RESOURCE: get_account_arn("ec2", "security-group/*"),
+            },
+            {
+                IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                IAMStatementProperty.ACTION: ["ec2:DescribeInstances", "ec2:DescribeRoutesTables"],
+                IAMStatementProperty.RESOURCE: "*",
+            },
+        ],
+        "ResourceType": ResourceType.EMR_CLUSTER,
+    },
+}
 
-    new_configuration = SettingsModel.from_dict(request["configuration"])
+# Instead of deleting the policy (which requires unattaching from all roles) assign a filler non-impactful statement
+filler_deny_statement = (
+    {
+        IAMStatementProperty.EFFECT: IAMEffect.DENY,
+        IAMStatementProperty.ACTION: ["dynamodb:CreateTable"],
+        IAMStatementProperty.RESOURCE: get_account_arn("dynamodb", "table/invalid-table"),
+    },
+)
 
-    app_configuration = AppConfigurationModel(
-        configScope=configScope,
-        version_id=version_id,
-        configuration=new_configuration,
-        changed_by=event["requestContext"]["authorizer"]["principalId"],
-        change_reason=request["changeReason"],
-        created_at=time.time(),
-    )
 
+@event_wrapper
+def update_configuration(event, context):
     try:
-        # Create a record for this configuration and mark it the latest config by incrementing the version Id
-        app_configuration_dao.create(config=app_configuration)
-    except Exception as e:
-        if e.response["Error"]["Code"] == "ConditionalCheckFailedException":
-            e.response["ResponseMetadata"]["HTTPStatusCode"] = 429
-        raise e
+        response_status_code = 200
+        warning_issues = []
+        request = json.loads(event["body"])
+        configScope = request["configScope"]
+        version_id = request["versionId"] + 1  # increment so this will be the latest version
+
+        new_configuration = SettingsModel.from_dict(request["configuration"])
+
+        app_configuration = AppConfigurationModel(
+            configScope=configScope,
+            version_id=version_id,
+            configuration=new_configuration,
+            changed_by=event["requestContext"]["authorizer"]["principalId"],
+            change_reason=request["changeReason"],
+            created_at=time.time(),
+        )
+
+        try:
+            # Create a record for this configuration and mark it the latest config by incrementing the version Id
+            app_configuration_dao.create(config=app_configuration)
```

Comment:
We should probably move all of this logic behind the IAM policy updates; as this is written, we could update the app config, and *then* attempt to update the policy, fail, and have an erroneous config.

Reply:
Agreed, I have the policy updates happen before the config is persisted in my PR

Reply:
Sounds good, I can do that.